src/abstractdataframe/abstractdataframe.jl

```diff
@@ -1650,22 +1662,44 @@ julia> flatten(df2, :b)
 │ 3   │ 2     │ r      │
 │ 4   │ 2     │ s      │
 
+julia> df3 = DataFrame(a = [1, 2], b = [[1, 2], [3, 4]], c = [[5, 6], [7]])
+2×3 DataFrame
+│ Row │ a     │ b      │ c      │
+│     │ Int64 │ Array… │ Array… │
+├─────┼───────┼────────┼────────┤
+│ 1   │ 1     │ [1, 2] │ [5, 6] │
+│ 2   │ 2     │ [3, 4] │ [7]    │
+
+julia> flatten(df3, [:b, :c])
+ERROR: ArgumentError: Vector lengths across columns in col, within the same row, must be the same
 ```
 """
-function flatten(df::AbstractDataFrame, col::ColumnIndex)
-    col_to_flatten = df[!, col]
-    lengths = length.(col_to_flatten)
-    new_df = similar(df[!, Not(col)], sum(lengths))
+function flatten(df::AbstractDataFrame,
+                 cols::Union{ColumnIndex, AbstractVector, Regex, Not, Between, All, Colon})
+    idxcols = index(df)[cols]
+    col1 = first(idxcols)
+    lengths = length.(df[!, col1])
+    for col in idxcols
+        v = df[!, col]
+        if any(x -> length(x[1]) != x[2], zip(v, lengths))
+            r = findfirst(!=(0), length.(v) .- lengths)
+            throw(ArgumentError("Lengths of iterables stored in columns $col1 and $col are not the the same in row $r"))
```

Comment:
can you please break this line (and I think also above one line also) so that it is not longer that 92 chars, as we try to stick to this limit. Thank you.